File: src/main/java/com/drunkensoftball/api/domain/BaseEntity.java
package com.drunkensoftball.api.domain;


import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.Calendar;
import java.util.UUID;

@MappedSuperclass
public abstract class BaseEntity {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "uuid")
    private String uuid;

    @Column(name = "created", updatable = false)
    private Calendar created;

    @Column(name = "updated")
    private Calendar updated;

    @JsonIgnore
    @Column(name = "removed", nullable=false)
    private boolean removed;

    @PrePersist
    protected void onCreate() {
        this.created = Calendar.getInstance();
        this.uuid = UUID.randomUUID().toString();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updated = Calendar.getInstance();
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Calendar getCreated() {
        return created;
    }

    public Calendar getUpdated() {
        return updated;
    }

    public void setUpdated(Calendar updated) {
        this.updated = updated;
    }

    public boolean isRemoved() {
        return removed;
    }

    public void setRemoved(boolean removed) {
        this.removed = removed;
    }

    public String getUuid() {
        return uuid;
    }
}

File: src/main/kotlin/com/drunkensoftball/api/domain/BaseEntity.kt
package com.drunkensoftball.api.domain


import com.fasterxml.jackson.annotation.JsonIgnore

import javax.persistence.*
import java.util.Calendar
import java.util.UUID

@MappedSuperclass
abstract class BaseEntity {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "uuid")
    var uuid: String? = null

    @Column(name = "created", updatable = false)
    var created: Calendar? = null

    @Column(name = "updated")
    var updated: Calendar? = null

    @JsonIgnore
    @Column(name = "removed", nullable = false)
    var removed: Boolean = false

    @PrePersist
    protected fun onCreate() {
        this.created = Calendar.getInstance()
        this.uuid = UUID.randomUUID().toString()
    }

    @PreUpdate
    protected fun onUpdate() {
        this.updated = Calendar.getInstance()
    }
}
